src/Configuration/BaseExtractorConfig.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorCommon\Configuration;
+
+use Keboola\Component\Config\BaseConfig;
+
+class BaseExtractorConfig extends BaseConfig
+{
+    /** @var DatabaseParameters|null */
+    private $databaseParameters;
+
+    public function getDbParameters(): DatabaseParameters
+    {
+        if ($this->databaseParameters) {
```

Comment:
Tohle je megapodezřelé. Není vůbec jasné, kde se tam ty database parameters vezmou a proč se neuloží... bývá takový ten pattern, že `if (!vytahnoutTo()) { ulozitTo() } vytahnoutTo()`, ale tady se to neuloží a dělá se to naopak...

Reply:
Tohle je prave kvuli tomu setteru, ktery se vola po nastaveni SSH. Kdyz by tam ten if nebyl, a setter to nastavil na jiny value objekt, tak by ti vracel stale `DatabaseParameters::fromRaw()`

Reply:
Myslím takhle:
```
getDbParameters() {
if (!$this->databaseParameters) {
  $this->databaseParameters = DatabaseParameters::fromRaw($this->getValude('db'));
}
return $this->databaseParameters;
}

$db = DatabaseParameters::fromRaw($otherArray);
$ob->setDbParameters($db);
$ob->getParameters() === $db //true
```

Reply:
Ale je to prostě úplně divné za mě. Dělá to nekonzistenci v tom objektu, kdy máš uložené hodnoty configu nějak ale vrací ti to DB parameters jinak - nějaké které to upeklo někde venku. Prostě to je takový code-smell IMHO.